megamek/src/megamek/common/LosEffects.java

```diff
@@ -1099,35 +1099,44 @@ private static LosEffects losForCoords(IGame game, AttackInfo ai,
             }
         }
         
-        // TODO: Identify when LOS travels *above* a building's hex.
-        // Alternatively, force all building hexes to be same height.
-
         // check for block by terrain
+        
+        
+        // All unit heights report as 1 less in MM than what they really are 
+        // (1 for mechs, 0 for tanks...)
+        // A level 4 hill will not block a mech on a level 3 hill
+        // (height of the mech in here = 3+"1" = "4"), as 
+        // hill elevation is not > unit elevation (normal LOS rules)
+        // With diagramming LOS it will block LOS as soon as the sightline
+        // drops by 0.1 to 3.9, even though that means it would be at 4.9 in
+        // "real" height values, so still well above the level 4 hill
+        // Therefore we need to add 1 from the diagramming LOS elevation
+        // to correct the calculation
+        // This is still hacky as Entity should simply report the real heights 
+        // and the comparison in here should follow TW/TO "higher or equal" rules.
+        
+        // The interpolated elevation for TacOps LOS diagramming
+        double losElevation = (ai.targetAbsHeight * ai.attackPos.distance(coords) 
+                + ai.attackAbsHeight * ai.targetPos.distance(coords))
+                    / (ai.targetPos.distance(coords) + ai.attackPos.distance(coords))
+                    + 1;
+        
+        // The higher of the attacker's height and defender's height
+        int maxUnitHeight = Math.max(ai.attackAbsHeight, ai.targetAbsHeight);
+        boolean attackerAdjc = ai.attackPos.distance(coords) == 1;
+        boolean targetAdjc = ai.attackPos.distance(coords) == 1;
```

Comment:
Did you mean to have this be equivalent to "attackerAdjc"? Also, what does Adjc stand for?

Reply:
Adjacent would be my bet

Reply:
Good you caught that! ai.targetPos indeed - wonder if I would have found that during testing.
And yes it's adjacent. Wanted to abbreviate it. Started with "adj", found it reads like adjustment, therefore the -c